CONTRIBUTING.md

```diff
@@ -1,47 +1,320 @@
+details about the experiment type, protocol, analysis, and implementation, which will
+give us the necessary information to decide whether the experiment is feasible to
+implement and useful to include in our package library.
+
+### Choosing an issue to work on
+We use the following labels to help non-maintainers find issues best suited to their
+interests and experience level:
+
+* [good first issue](https://github.com/Qiskit/qiskit-experiments/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+  - these issues are typically the simplest available to work on, perfect for newcomers.
+  They should already be fully scoped, with a clear approach outlined in the
+  descriptions.
+* [help wanted](https://github.com/Qiskit/qiskit-experiments/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
+  - these issues are generally more complex than good first issues. They typically cover
+  work that core maintainers don't currently have capacity to implement and may require
+  more investigation/discussion. These are a great option for experienced contributors
+  looking for something a bit more challenging.
 
 ### Pull request checklist
 
-When submitting a pull request and you feel it is ready for review,
-please ensure that:
-
-1. The code follows the code style of the project and successfully
-   passes the tests. For convenience, you can execute `tox` locally,
-   which will run these checks and report any issues.
-2. The documentation has been updated accordingly. In particular, if a
-   function or class has been modified during the PR, please update the
-   *docstring* accordingly.
-3. You update the relevant tutorial or write a new one. Use this [tutorial guidelines](docs/tutorials/GUIDELINES.md)
-   document for further details.
-
-   In case the PR needs to be merged without delay (e.g. for a high priority fix), open an issue for updating or adding
-   the tutorial later.
-4. If it makes sense for your change that you have added to include new tests that cover the changes.
-5. If your change has an end user facing impact (new feature,
-   deprecation, removal etc) that you have added a reno release note for that
-   change and that the PR is tagged for the changelog.
-
-### API documentation
-
-Qiskit Experiments provides a special syntax and macros as [Sphinx](https://www.sphinx-doc.org/en/master/) extensions
-to format experiment and analysis class docstring. If you implement new experiment or analysis,
-you should use following style so that documentations are formatted in the same manner throughout our experiment library.
-You can use standard [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html)
+When submitting a pull request for review, please ensure that:
+
+1. The code follows the code style of the project and successfully passes the tests.
+2. The API documentation has been updated accordingly.
+3. You have updated the relevant tutorial or write a new one. In case the PR needs to be
```

Comment:
Writing tutorial is not mandatory if we move to example-based tutorial rather than component-wise. But PR affects existing class API one should update corresponding tutorial.